articles/active-directory/develop/jwt-claims-customization.md

```diff
@@ -281,87 +281,46 @@ To run this script you need:
 > The private key must be in PKCS#12 format since Azure AD doesn't support other format types. Using the wrong format can result in the error "Invalid certificate: Key value is invalid certificate" when using Microsoft Graph to PATCH the service principal with a `keyCredentials` containing the certificate information.
 
 ```
-$fqdn="fourthcoffeetest.onmicrosoft.com" # this is used for the 'issued to' and 'issued by' field of the certificate
-$pwd="mypassword" # password for exporting the certificate private key
-$location="C:\\temp" # path to folder where both the pfx and cer file will be written to
+##########################################################
+# Replace the variables below with the appropriate values
+
+$fqdn="yourDomainHere" # This is used for the 'issued to' and 'issued by' field of the certificate
+$pwd="myPassword" # password for exporting the certificate private key
+$tenantId   = "tenantId" # Replace with Tenant ID
+$appObjId = "objectID" # Replace with the Object ID of the App Registration
+
+#
+##########################################################
 
 # Create a self-signed cert
 $cert = New-SelfSignedCertificate -certstorelocation cert:\currentuser\my -DnsName $fqdn
 $pwdSecure = ConvertTo-SecureString -String $pwd -Force -AsPlainText
 $path = 'cert:\currentuser\my\' + $cert.Thumbprint
+$location="C:\\temp" # path to folder where both the pfx and cer file will be written to
 $cerFile = $location + "\\" + $fqdn + ".cer"
 $pfxFile = $location + "\\" + $fqdn + ".pfx"
- 
+
 # Export the public and private keys
 Export-PfxCertificate -cert $path -FilePath $pfxFile -Password $pwdSecure
 Export-Certificate -cert $path -FilePath $cerFile
 
-$ClientID = "<app-id>"
-$loginURL       = "https://login.microsoftonline.com"
-$tenantdomain   = "fourthcoffeetest.onmicrosoft.com"
-$redirectURL = "http://localhost" # this reply URL is needed for PowerShell Core 
-[string[]] $Scopes = "https://graph.microsoft.com/.default"
 $pfxpath = $pfxFile # path to pfx file
 $cerpath = $cerFile # path to cer file
-$SPOID = "<service-principal-id>"
-$graphuri = "https://graph.microsoft.com/v1.0/serviceprincipals/$SPOID"
 $password = $pwd  # password for the pfx file
- 
- 
-# choose the correct folder name for MSAL based on PowerShell version 5.1 (.Net) or PowerShell Core (.Net Core)
- 
+
+# Check PowerShell version 5.1 (.Net) or PowerShell Core (.Net Core) and read the certificate file accordingly
```

Comment:
@franciscopgomes, is Powershell v5.1 a minimum version here or the exact version of 5.1 needed? Does this cover all powershell versions 5.1+?

Reply:
@cliwerner this was already present in the previous version of the script (which I didn't own) to account for a difference in folder structuring for different PowerShell versions. Based on tests, it did cover both v5.1 and other PS versions.